application/src/test/java/org/togetherjava/tjbot/commands/mediaonly/MediaOnlyChannelListenerTest.java

```diff
@@ -1,86 +1,62 @@
-import java.util.List;
 
-import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.*;
 
 class MediaOnlyChannelListenerTest {
 
-    private JDA jda;
-    private final Message message = mock(Message.class);
-    private final long responseNumber = 1L;
-    private final User user = mock(User.class);
+    JdaTester jdaTester;
+    private Message message;
+    private User user;
     private MediaOnlyChannelListener mediaOnlyChannelListener;
 
     @BeforeEach
     void setUp() {
-        JdaTester jdaTester = new JdaTester();
-        jda = jdaTester.getJdaMock();
-
+        jdaTester = new JdaTester();
+        message = jdaTester.createMessageSpy();
+        Member memberSpy = jdaTester.createMemberSpy(1L);
+        user = memberSpy.getUser();
         Config config = mock(Config.class);
         when(config.getMediaOnlyChannelPattern()).thenReturn("memes");
         mediaOnlyChannelListener = new MediaOnlyChannelListener(config);
     }
 
     @Test
     void validMessagePostWithAttachment() {
-        Message.Attachment attachment = new Message.Attachment(1L, null, null, "TEST",
-                "Test ContentType", null, 1, 1, 1, false, (JDAImpl) jda);
-        List<Message.Attachment> attachments = Collections.singletonList(attachment);
-
-        when(message.getAuthor()).thenReturn(user);
-        when(message.getAttachments()).thenReturn(attachments);
-
+        Message.Attachment attachment = jdaTester.createAttachment(1L, null, null, "TEST",
+                "Test ContentType", null, 1, 1, 1, false);
+        jdaTester.mockMessageAttachments(message, Collections.singletonList(attachment));
         MessageReceivedEvent messageReceivedEvent =
-                new MessageReceivedEvent(jda, responseNumber, message);
+                jdaTester.createMessageReceivedEvent(message, user);
         mediaOnlyChannelListener.onMessageReceived(messageReceivedEvent);
         verify(message).getAttachments();
     }
 
     @Test
     void validMessagePostWithUrlEmbedded() {
-        MessageEmbed messageEmbed = new MessageEmbed("https://9gag.com/gag/a61A238", "Test", "Test",
-                EmbedType.LINK, null, 1, null, null, null, null, null, null, null);
-        List<MessageEmbed> messageEmbeds = Collections.singletonList(messageEmbed);
-
-        when(message.getAuthor()).thenReturn(user);
-        when(message.getEmbeds()).thenReturn(messageEmbeds);
-
+        MessageEmbed messageEmbed = jdaTester.createMessageEmbed("https://9gag.com/gag/a61A238",
+                "Test", "Test", EmbedType.LINK, null, 1, null, null, null, null, null, null, null);
+        jdaTester.mockMessageEmbeds(message, Collections.singletonList(messageEmbed));
         MessageReceivedEvent messageReceivedEvent =
-                new MessageReceivedEvent(jda, responseNumber, message);
+                jdaTester.createMessageReceivedEvent(message, user);
         mediaOnlyChannelListener.onMessageReceived(messageReceivedEvent);
```

Comment:
This piece of code is duplicated in all ur tests. Id suggest making a private helper method in this test class. Sth along the lines of `private void sendMessage(emebeds)` which then does:
```java
jdaTester.mockMessageEmbeds(message, embeds);
MessageReceivedEvent messageReceivedEvent = jdaTester.createMessageReceivedEvent(message, user);
mediaOnlyChannelListener.onMessageReceived(messageReceivedEvent);
```
or similar